File: python-systempromptleakage/vulnerable11.py
from rest_framework.decorators import api_view
from rest_framework.response import Response
from openai import OpenAI

client = OpenAI()
system_prompt = "You are a product recommendation engine. Internal pricing tiers: A=$100, B=$50, C=$20."

@api_view(["POST"])
def ask(request):
    message = request.data.get("message")
    response = client.chat.completions.create(
        model="gpt-4o",
        max_tokens=1024,
        messages=[
            {"role": "system", "content": system_prompt},
            {"role": "user", "content": message}
        ]
    )
    return Response({"answer": response.choices[0].message.content, "prompt_used": system_prompt})

File: typescript-unboundedconsumption/vulnerable11.ts
import OpenAI from "openai";

const openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

async function processUserRequest(
    userMessage: string,
    adminMessage: string
): Promise<{ analysis: string; summary: string }> {
    const analysisResult = await openai.chat.completions.create({
        model: "gpt-4o-mini",
        messages: [{ role: "user", content: adminMessage }],
        max_tokens: 100,
    });

    const summary = await openai.chat.completions.create({
        model: "gpt-4o",
        messages: [
            { role: "system", content: "Summarize the following in detail." },
            { role: "user", content: userMessage },
        ],
    });

    return {
        analysis: analysisResult.choices[0].message.content ?? "",
        summary: summary.choices[0].message.content ?? "",
    };
}

export { processUserRequest };
